src/core/Microsoft.PowerToys.Settings.UI.Lib/ViewModels/FancyZonesViewModel.cs

```diff
@@ -121,9 +104,10 @@ public bool IsEnabled
                 if (value != _isEnabled)
                 {
                     _isEnabled = value;
-                    GeneralSettings generalSettings = _settingsUtils.GetSettings<GeneralSettings>(string.Empty);
-                    generalSettings.Enabled.FancyZones = value;
-                    OutGoingGeneralSettings snd = new OutGoingGeneralSettings(generalSettings);
+
+                    // Set the status of FancyZones in the general settings configuration
+                    GeneralSettingsConfig.Enabled.FancyZones = value;
```

Comment:
I was hoping that we were going to encapsulate this in the settings repository/settingsutils.   Without this, how does someone know when it's appropriate to use IPC as in this case vs SaveSettings.    Also to truly fix this issue we may need to synchronize write / reads across the IPC boundary.  We would have to do this in a bunch of different places.

Reply:
@ryanbodrug-microsoft, these are the changes that I've made since you last reviewed-
1. unify the way the Modulename is accessed. It was redeclared in multiple places and this would cause an issue if the name is changed only in one place. All the module names are accessed using the <T>Settings.ModuleName, eg: ShortcutGuideSettings.ModuleName.
2. Modified the settingsConfig in settingsRepository so that the we don't have to use TypeOf and categorize the objects into GeneralSettings and BasePTModuleSettings.
3. GetFile is now a private function. Modified the logic of KBM default.json access and PT Run so that we can re-use GetSettings instead of GetFile.
4. Added UpgradeSettingsConfiguration to the ISettingsConfig interface so that the settings file can be upgraded based on some condition. Presently, only the GeneralSettings file is utilizing this to change the PT Version number based on the old PT version and the current PT version that it receives from the helper function. Verified that if the PT version is lower in the general settings.json file, settings saves the file with the new version info.
5. The naming for the PowerToys was inconsistent and the variables were redeclared in multiple places. To have the settings.ModuleName as the main name, all other places should refer to that name. In the tests file the module name for ImgResizer was "ImageResizer" and not "Image Resizer".

As discussed offline, there is still some refactoring that needs to be done for settings, however it would be nice to get this in for this release as it fixes the crash and I'll make the following changes as a part of another PR -
1. Move IPC from the shell page to settingsUtils.
2. Have the viewmodels utilize the settings utils to communicate with runner rather than through having a delegate passed to each of them.
3. We should be able to make the code such that settings doesn't have to save the file directly, it can rather communicate this information to the runner and the runner can save the information. 
4. As stated previously, only FZ, Shortcutguide, and powerPreview use settings repository. It should be expanded to the rest so that ideally `SaveSettings` is made private.
5. If we can somehow have runner tell the settings that it has received the information and has successfully stored it, then we can notify the user if something goes wrong there instead of having settings and runner running completely independently. I need to look more into the way IPC works to do this..